src/framework/dropdown/DropdownGroup.jsx

```diff
@@ -0,0 +1,101 @@
+
+import React, {
+  Component,
+  PropTypes,
+} from 'react';
+import classNames from 'classnames';
+
+import BaseDropdown from '../base/dropdown/BaseDropdown.jsx';
+import DropdownDot from './dropdownDot/DropdownDot.jsx';
+import DropdownOption from './DropdownOption.jsx';
+
+export {
+  default as DropdownDot,
+} from './dropdownDot/DropdownDot.jsx';
+
+export {
+  default as DropdownOption,
+} from './DropdownOption.jsx';
+
+export default class DropdownGroup extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.labelProvider = this.labelProvider.bind(this);
+
+    this.dropdownDotColorToClassesMap = {
+      [DropdownDot.COLOR.BLUE]: 'dropdownLabel--blue',
+      [DropdownDot.COLOR.GREEN]: 'dropdownLabel--green',
+      [DropdownDot.COLOR.GREY]: 'dropdownLabel--grey',
+      [DropdownDot.COLOR.RED]: 'dropdownLabel--red',
+    };
+  }
+
+  labelProvider(option) {
+    let label = this.props.labelProvider(option);
+
+    if (this.props.dotColor) {
+      label = [
+        <DropdownDot
+          color={this.props.dotColor}
+          key={0}
+        />,
+        <span
+          className={this.dropdownDotColorToClassesMap[this.props.dotColor]}
+          key={1}
+        >
+          {label}
+        </span>,
+      ];
+    }
+
+    return label;
+  }
+
+  render() {
+    const labelClasses = classNames(this.props.labelClasses, {
+      'dropdownLabel--borderless': this.props.isBorderless,
+      'dropdownLabel--readonly': this.props.isReadonly,
+    });
+
+    const optiontolist = this.props.options.reduce(
```

Comment:
Camel case. Plus the result is the list itself (like `options` or `optionList`), but `optionToList` indicates a method rather than a list I think.